diff -urN uClibc-0.9.27/ldso/ldso/dl-elf.c uClibc-0.9.27-new/ldso/ldso/dl-elf.c
--- uClibc-0.9.27/ldso/ldso/dl-elf.c	2005-01-11 23:59:21.000000000 -0800
+++ uClibc-0.9.27-new/ldso/ldso/dl-elf.c	2007-02-09 16:20:14.000000000 -0800
@@ -148,11 +148,13 @@
 		return NULL;
 
 	/* We need a writable copy of this string */
-	path = _dl_strdup(path_list);
+	path = alloca(_dl_strlen(path_list));
+
 	if (!path) {
 		_dl_dprintf(2, "Out of memory!\n");
 		_dl_exit(0);
 	}
+	_dl_strcpy(path, path_list);
 
 
 	/* Unlike ldd.c, don't bother to eliminate double //s */
@@ -172,9 +174,7 @@
 		_dl_strcat(mylibname, "/");
 		_dl_strcat(mylibname, name);
 		if ((tpnt1 = _dl_load_elf_shared_library(secure, rpnt, mylibname)) != NULL)
-		{
 			return tpnt1;
-		}
 		path_n += (_dl_strlen(path_n) + 1);
 	}
 	return NULL;
diff -urN uClibc-0.9.27/Rules.mak uClibc-0.9.27-new/Rules.mak
--- uClibc-0.9.27/Rules.mak	2005-01-11 23:59:21.000000000 -0800
+++ uClibc-0.9.27-new/Rules.mak	2007-02-09 16:26:28.000000000 -0800
@@ -285,7 +285,7 @@
     CFLAGS += $(PICFLAG)
 endif
 
-ASFLAGS = $(CFLAGS)
+ASFLAGS = $(CFLAGS) -D__ASSEMBLY__
 ifeq ($(UCLIBC_BUILD_NOEXECSTACK),y)
 check_as_noexecstack=$(shell if $(LD) --help | grep -q "z noexecstack"; then echo "-Wa,--noexecstack"; fi)
 ASFLAGS += $(check_as_noexecstack)
