diff -Nur uClibc-0.9.27.old/libpthread/linuxthreads/semaphore.c uClibc-0.9.27/libpthread/linuxthreads/semaphore.c
--- uClibc-0.9.27.old/libpthread/linuxthreads/semaphore.c	2006-04-21 16:05:56.519478548 -0700
+++ uClibc-0.9.27/libpthread/linuxthreads/semaphore.c	2006-04-21 16:03:48.128558709 -0700
@@ -226,7 +226,8 @@
     /* The standard requires that if the function would block and the
        time value is illegal, the function returns with an error.  */
     __pthread_unlock(&sem->__sem_lock);
-    return EINVAL;
+    __set_errno (EINVAL);
+    return -1;
   }
 
   /* Set up extrication interface */
@@ -264,7 +265,8 @@
 
 	if (was_on_queue) {
 	  __pthread_set_own_extricate_if(self, 0);
-	  return ETIMEDOUT;
+	  __set_errno (ETIMEDOUT);
+	  return -1;
 	}
 
 	/* Eat the outstanding restart() from the signaller */
