diff -urN uClibc-0.9.28.3/Rules.mak uClibc-0.9.28.3-modified/Rules.mak
--- uClibc-0.9.28.3/Rules.mak	2007-02-28 13:12:06.000000000 -0800
+++ uClibc-0.9.28.3-modified/Rules.mak	2007-08-03 14:44:14.000000000 -0700
@@ -167,7 +167,7 @@
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_2)+=-mips2 -mtune=mips2
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_3)+=-mips3 -mtune=mips3
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_4)+=-mips4 -mtune=mips4
-	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS32)+=-mips32 -mtune=mips32
+	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS32)+=-mips32r2 -mtune=mips32r2 -Wa,-mips32r2
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS64)+=-mips64 -mtune=mips32
 endif
 
